f5_openstack_agent/lbaasv2/drivers/bigip/pool_service.py

```diff
@@ -54,6 +54,71 @@ def __init__(self, service_adapter):
         self.node_helper = resource_helper.BigIPResourceHelper(
             resource_helper.ResourceType.node)
 
+    def delete_gateway_pool(self, bigip, pool_name, partition='Common'):
```

Comment:
I see too much generic resource create/delete procedure. I think we do need to consolidate them into a generic super class.